packages/zendesk-adapter/src/filters/help_center_service_url.ts

```diff
@@ -0,0 +1,97 @@
+  GUIDE_SETTINGS_TYPE_NAME, BRAND_TYPE_NAME,
+} from '../constants'
+
+const PARAM_MATCH = /\{([\w_.]+)}/g
+const DIFFERENT_BASE_URL_TYPES = [CATEGORY_TRANSLATION_TYPE_NAME, GUIDE_SETTINGS_TYPE_NAME]
+const SERVICE_URL_FOR_GUIDE: Record<string, string> = {
+  [SECTION_TYPE_NAME]: '/knowledge/arrange/sections/{id}?brand_id={brand}',
+  [CATEGORY_TYPE_NAME]: '/knowledge/arrange/categories/{id}?brand_id={brand}',
+  [ARTICLE_TYPE_NAME]: '/knowledge/articles/{id}/{source_locale}?brand_id={brand}',
+  [ARTICLE_TRANSLATION_TYPE_NAME]: '/knowledge/articles/{_parent.0.value.value.id}/{locale}?brand_id={brand}',
+  [SECTION_TRANSLATION_TYPE_NAME]: '/knowledge/sections/{_parent.0.value.value.id}?brand_id={brand}&locale={locale}',
+  [CATEGORY_TRANSLATION_TYPE_NAME]: '/hc/admin/categories/{_parent.0.value.value.id}/edit?translation_locale={locale}',
+  [GUIDE_SETTINGS_TYPE_NAME]: '/hc/admin/general_settings',
+}
+
+const replaceUrlParamsBrand = (url: string, instance: InstanceElement): string =>
+  url.replace(
+    PARAM_MATCH,
+    val => {
+      let replacement
+      if (val.slice(1, -1).startsWith('_')) { // meaning that it refers to an annotation
+        replacement = _.get(instance.annotations, val.slice(1, -1)) ?? val
+      } else {
+        replacement = instance.value[val.slice(1, -1)] ?? val
+      }
+      if (!isPrimitiveValue(replacement)) {
+        throw new Error(`Cannot replace param ${val} in ${url} with non-primitive value ${replacement}`)
+      }
+      return replacement.toString()
+    }
+  )
+
+const createServiceUrl = (instance: InstanceElement, baseUrl:string): void => {
+  const url = replaceUrlParamsBrand(SERVICE_URL_FOR_GUIDE[instance.elemID.typeName], instance)
+  instance.annotations[CORE_ANNOTATIONS.SERVICE_URL] = (new URL(url, baseUrl)).href
+}
+
+/**
+ * this filter creates the serviceUrl annotation for guide elements. It first determines the baseUrl
+ * (category_translation and guide_settings have different ones depending on the brand). Then the
+ * filter replaces url params and creates the service url accordingly.
+ */
+const filterCreator: FilterCreator = ({ client }) => ({
+  onFetch: async (elements: Element[]): Promise<void> => {
+    const brandList = elements
+      .filter(isInstanceElement)
+      .filter(instance => instance.elemID.typeName === BRAND_TYPE_NAME)
+      .map(brand => brand.value)
+    const brandToUrl: Record<number, string> = _.mapValues(_.keyBy(brandList, 'id'), 'brand_url')
+
+    elements
+      .filter(isInstanceElement)
+      .filter(instance => Object.keys(SERVICE_URL_FOR_GUIDE).includes(instance.elemID.typeName))
+      .forEach(instance => {
+        let baseUrl
+        // only some have different baseUrl according to brand
+        if (DIFFERENT_BASE_URL_TYPES.includes(instance.elemID.typeName)) {
+          baseUrl = brandToUrl[instance.value.brand]
+        } else {
+          baseUrl = client.getUrl().href
+        }
+        createServiceUrl(instance, baseUrl)
+      })
```

Comment:
nit: maybe avoid the `let` by having a function that returns the base url
for example, you can create a function in the beginning of `onFetch` that calculates the base url by brand:

```suggestion
    const toBaseUrl = (instance: InstanceElement): string => {
      // only some have different baseUrl according to brand
      if (DIFFERENT_BASE_URL_TYPES.includes(instance.elemID.typeName)) {
        return brandToUrl[instance.value.brand]
      }
      return client.getUrl().href
    }

    elements
      .filter(isInstanceElement)
      .filter(instance => Object.keys(SERVICE_URL_FOR_GUIDE).includes(instance.elemID.typeName))
      .forEach(instance => {
        createServiceUrl(instance, toBaseUrl(instance))
      }
```